Simplenote/MarkdownViewController.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+import WebKit
+
+
+// MARK: - MarkdownViewController
+//
+@objcMembers
+class MarkdownViewController: NSViewController {
+
+    /// Main WebView
+    ///
+    @IBOutlet private var webView: WKWebView!
+
+    /// Allowed Outgoing link Schemes
+    ///
+    private let allowedOutboundSchemes = ["http", "https", "mailto"]
+
+    /// Markdown Text to be rendered
+    ///
+    var markdown: String? {
+        didSet {
+            reloadHTML()
+        }
+    }
+
+
+    // MARK: - Lifecycle
+
+    deinit {
+        stopListeningToNotifications()
+    }
+
+    override init(nibName nibNameOrNil: NSNib.Name?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        startListeningToNotifications()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        startListeningToNotifications()
+    }
+
+
+    /// For performance purposes: We'll ensure the WebView is ready to refresh in a split second
+    ///
+    func preloadView() {
+        if isViewLoaded {
+            return
+        }
+
+        loadView()
+        reloadHTML()
+    }
+}
+
+
+// MARK: - WKNavigationDelegate
+//
+extension MarkdownViewController: WKNavigationDelegate {
+
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        webView.isHidden = false
+    }
+
+    func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
+        webView.isHidden = false
+    }
+
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        guard navigationAction.navigationType == .linkActivated else {
+            decisionHandler(.allow)
+            return
+        }
+
+        if let url = navigationAction.request.url, let scheme = url.scheme?.lowercased(), allowedOutboundSchemes.contains(scheme) {
+            NSWorkspace.shared.open(url)
+        }
+
+        decisionHandler(.cancel)
+    }
+}
+
+
+// MARK: - Notification Helpers
+//
+extension MarkdownViewController {
+
+    func startListeningToNotifications() {
+        NotificationCenter.default.addObserver(self, selector: #selector(reloadHTML), name: .VSThemeManagerThemeDidChange, object: nil)
+    }
+
+    func stopListeningToNotifications() {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    @objc
+    func reloadHTML() {
+        let content = markdown ?? ""
+        let html = SPMarkdownParser.renderHTML(fromMarkdownString: content) ?? ""
+
+        // Workaround: Prevents UI flashes by hiding / unhiding when rendering is done
+        webView.isHidden = true
```

Comment:
Have you tried `suppressesIncrementalRendering` from `WKWebViewConfiguration`?
Not sure if it will solve the problem but seems like the effect should be similar 🤔

Reply:
**OH MY GOD THANK YOU THANK YOU!!!!!**